source/beamline/BioXAS/BioXASZebraPulseControl.cpp

```diff
@@ -167,6 +152,46 @@ bool BioXASZebraPulseControl::outputValue() const
 	return int(outputPulseControl_->value()) == 1;
 }
 
+bool BioXASZebraPulseControl::validTimeValue(double timeValue) const
+{
+	bool result = false;
+
+	if (timeValue >= BIOXASZEBRAPULSECONTROL_PULSE_WIDTH_MIN && timeValue <= BIOXASZEBRAPULSECONTROL_PULSE_WIDTH_MAX)
+		result = true;
+
+	return result;
+}
+
+double BioXASZebraPulseControl::convertTimeValue(double timeValue, double timeUnits, double desiredTimeUnits) const
```

Comment:
This may be a convenience thing. The timeUnitsControl_ will always report its value as a double. Arguments that are doubles mean the function will accept arguments that are in double, int, or enum form--it's the most flexible.